### ISSUE TITLE
Move fire maps from surface dataset to stream files

### ISSUE THREAD

**Issue by samsrabin** [ESCOMP/CTSM#2617]
- `peatf` (peatland fraction)
- `abm` (agricultural fire peak month)

I will need @lifang0209 to point me to the highest-resolution versions of these files that she has.

**Comment by ekluzek** [comment]
As a starting point you should probably start with the half degree datasets being used by mksurfdata_esmf. That'll provide a nice transition from what's currently being done, to new higher resolution datasets to be used in the future.

Is this change really needed for the Capability freeze? I suppose it might so we don't have to make new surface datasets if updated datasets are required. This might be something good to discuss, so I'll add the next label to it...

**Comment by samsrabin** [comment]
It's only needed for the capability freeze because it needs to happen before I bring in new datasets from Fang.

**Comment by wwieder** [comment]
These fields should be removed from the 5.3 datasets too, @lawrencepj1 and @slevis-lmwg.

**Comment by ekluzek** [comment]
We will include the new data on the ctsm5.3.0 surface datasets. But, after ctsm5.3.0 this change will come in which will enable clm4_5 and clm5_0 to be closer to what they were previous ctsm5.3.0.

**Comment by samsrabin** [comment]
These will be available on the half-degree surface datasets produced from #2500.

**Comment by lifang0209** [comment]
The variable abm represents the crop peak month, with 1 for Jan, 2 for Feb, ..., and 12 for Dec. For a 1-degree grid, if the values are 0, 0, 1, 7 for each 0.5-degree grid, the result becomes 2 (Feb) using your averaging method, which is incorrect. In this case, assigning a value of 0 (i.e., mode) would be more reasonable.

I’m not sure if the regridding method in CLM includes using the mode (i.e., the most common value). If not, a simpler solution would be that I provide the abm raw data at 1.9x2.5 resolution, where higher-resolution grids adopt the value of the corresponding 1.9x2.5 grid they fall in.

If both are hard for CLM, then the simple method is to add an IF statement in the fire model to assume no crop fire if crop coverage is less than 10%.

**Comment by ekluzek** [comment]
@slevis-lmwg in answer to your question at the end here...

> * abm is "agricultural fire peak month" and should not be spatially averaged from the raw data to the fsurdat file
> * Instead we should pick the dominant value when interpolating
> * We are not doing this correctly in mksurfdata_esmf and this is the issue that I plan to open

> * And my question: Could this be done correctly if abm were changed to a stream, which is why I'm bringing up the topic here

The same regridding options are available to us for a stream as are available in mksurfdata_esmf, because we are using ESMF methods. The ESMF methods are traditional spatial averaging. So there isn't an option to include the dominant value. It *could* be done in streams, but at this point it's likely to be easier to do it in mksurfdata_esmf.

**Comment by wwieder** [comment]
@slevis-lmwg will create a new issue for how we'll handle abm more robustly with the 5.4 datasets
